File: src/tools/users_snapshot/userUtils.ts
import {log} from "../../utils/mcpLogger.js";
import {LoggingLevelSchema} from "@modelcontextprotocol/sdk/types.js";

export function isValidResponse(accountId: string, response: any): boolean {

    if (response.data?.accountSnapshot?.users) {
        return true;
    }

    log(LoggingLevelSchema.Enum.debug, `No users found in account snapshot for remote_users_details account ID: ${accountId}`);
    return false;
}

export function emptyUsersResponse(timestamp: string): any {
    return {
        "data": {
            "accountSnapshotTimestamp": timestamp,
            "users": []
        },
    }
}

File: python/src/tools/users_snapshot/user_utils.py
"""
User utilities
"""
from typing import Any, Dict
from ...utils.mcp_logger import log
from mcp import LoggingLevel  # type: ignore[import-untyped]


def is_valid_response(account_id: str, response: Dict[str, Any]) -> bool:
    """
    Validate response for user snapshot queries.
    
    Args:
        account_id: Account ID
        response: Response dictionary
        
    Returns:
        True if valid, False otherwise
    """
    if response.get("data", {}).get("accountSnapshot", {}).get("users"):
        return True
    
    log("debug", f"No users found in account snapshot for remote_users_details account ID: {account_id}")
    return False


def empty_users_response(timestamp: str) -> Dict[str, Any]:
    """
    Create empty users response.
    
    Args:
        timestamp: Timestamp string
        
    Returns:
        Empty users response
    """
    return {
        "data": {
            "accountSnapshotTimestamp": timestamp,
            "users": []
        },
    }

